sympy/polys/polytools.py

```diff
@@ -5622,12 +5622,12 @@ def _sorted_factors(factors, method):
         def key(obj):
             poly, exp = obj
             rep = poly.rep.rep
-            return (exp, len(rep), rep)
+            return (exp, len(rep), len(poly.gens), rep)
```

Comment:
this needs a test that fails in master but works here; perhaps the same expr but with `method=sqf`? 

The live.sympy.org site is down right now @asmeurer with message: "You have reached a domain that is pending ICANN verification."